workflow/rocoto/gfs_tasks.py

```diff
@@ -570,13 +570,17 @@ def prepsnowcover(self):
         deps.append(rocoto.add_dependency(dep_dict))
         dependencies = rocoto.create_dependency(dep=deps)
 
+        cycledef = 'gdas_prep_snocvr'
+        if self.run in ['gfs']:
+            cycledef = self.run
+
         resources = self.get_resource('prepsnowcover')
         task_name = f'{self.run}_prepsnowcover'
         task_dict = {'task_name': task_name,
                      'resources': resources,
                      'dependency': dependencies,
                      'envars': self.envars,
-                     'cycledef': 'gdas_prep_snocvr',
+                     'cycledef': cycledef,
```

Comment:
I think this solution will only work if the gfs runs on just the 00z cycle.  If the gfs forecast is to run on 06, 12, and/or 18z, then this job will be added to the mesh and will fail.  I see two options.

1. Add another `cycledef`: `gfs_prep_snocvr`.  The `cycledef` could then be defined here as `f"{run}_prep_snocvr"`
2. Add an if block:
(pseudocode)
```pseudocode
if run == 'gfs':
   <add a taskvalid dependency for gdas_prepsnowcover>
   <cycledef = 'gfs'>
else:
   <cycledef = 'gdas_prep_snocvr'>
```

Reply:
Thanks @DavidHuber-NOAA for your review. Per above test, the gfs runs at 18z cycle, the added `gfs_prepsnowcover` job did nothing and will not stall the workflow. If the gfs forecast is to run on 06, 12, and/or 18z, the script `scripts/exglobal_prep_snowcover.py` will control `gfs_prepsnowcover` job doing nothing as below: 
```
    if SnowAnl.task_config.cyc == 0:
        SnowAnl.prepare_IMS()
```
Therefore, the current settings won't cause any problems to the workflow. 

I was thinking to add another `cycledef`: `gfs_prep_snocvr`, but I don't think it helps, because the `gfs_prep_snocvr` should be the same as `gdas_prep_snocvr`. 

In addition, the option 2 you proposed above is the same with current setting as below: 

```
        cycledef = 'gdas_prep_snocvr'
        if self.run in ['gfs']:
            cycledef = self.run
```
@DavidHuber-NOAA Do you agree with me? Thank you very much.

Reply:
Yes, I do agree with you now that I see the change you made in `exglobal_prep_snowcover.py`.  I think we would prefer to not have do-nothing jobs, but this would work for now.

@WalterKolczynski-NOAA what do you think?

Reply:
Agree that we want to eliminate do-nothing jobs.

For a solution, we need to consider what should happen in all the cases where there is not a GFS run every cycle. Is the goal to have the GFS version once per day? Or should it really only run at 00z? Can it run every day at 00z even if there is no GFS otherwise in that cycle? What about if there is no GFS cycle on a given day? What is actually being done by this job for the gfs?

We had similar restraints with metp, and the answers were once a day and it could be run even if there was no GFS in that cycle, but only if there was GFS sometime that day. The solution there was to run metp at 18z every day if the GFS ran more frequently than once a day (even if GFS did not run at 18z), and on the gfs cycledef if the GFS was run less often than once/day.

Reply:
Thanks @WalterKolczynski-NOAA for your reply. I think @CoryMartin-NOAA is best person to give more accurate answers. 

@CoryMartin-NOAA Would you please answer these questions? Thanks.

Reply:
I'd need to check to confirm, but I think, while it doesn't need to run for GFS and would be identical to GDAS, the GFS cycle runs 'first', so in ops this matters, but in R&D not so much. So perhaps we can use the GDAS 00z file in the GFS 00z analysis (but I don't know if this dependency is a good idea).

For your other questions:
- only at 00z, it's based on a L3 product
- yes, waste of resources but no harm otherwise
- either can skip it completely or run and produce a useless file
- See above for answer, it's identical, but dependencies could be tricky